packages/web/src/components/artist/ArtistChipTips.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { useEffect, useState } from 'react'
+
+import { IconTrophy, IconTrending } from '@audius/stems'
+import cn from 'classnames'
+
+import { ReactComponent as IconTip } from 'assets/img/iconTip.svg'
+import { useSelector } from 'common/hooks/useSelector'
+import { ID } from 'common/models/Identifiers'
+import { StringWei } from 'common/models/Wallet'
+import { getProfileUser } from 'common/store/pages/profile/selectors'
+import { getSupporters, getSupporting } from 'common/store/tipping/selectors'
+import { Nullable } from 'common/utils/typeUtils'
+import { formatWei, stringWeiToBN } from 'common/utils/wallet'
+import { USER_LIST_TAG as SUPPORTING_USER_LIST_TAG } from 'pages/supporting-page/sagas'
+import { USER_LIST_TAG as TOP_SUPPORTERS_USER_LIST_TAG } from 'pages/top-supporters-page/sagas'
+import { TIPPING_TOP_RANK_THRESHOLD } from 'utils/constants'
+import { encodeHashId } from 'utils/route/hashIds'
+
+import styles from './ArtistChip.module.css'
+
+const messages = {
+  audio: '$AUDIO',
+  supporter: 'Supporter'
+}
+
+const TIP_SUPPORT_TAGS = new Set([
+  SUPPORTING_USER_LIST_TAG,
+  TOP_SUPPORTERS_USER_LIST_TAG
+])
+
+type ArtistChipTipsProps = {
+  userId: ID
+  tag: string
+}
+
+export const ArtistChipTips = ({ userId, tag }: ArtistChipTipsProps) => {
+  const encodedUserId = encodeHashId(userId)
+  const profile = useSelector(getProfileUser)
+  const encodedProfileUserId = encodeHashId(profile?.user_id ?? null)
+  const supportingMap = useSelector(getSupporting)
+  const supportersMap = useSelector(getSupporters)
+  const [amount, setAmount] = useState<Nullable<StringWei>>(null)
+  const [rank, setRank] = useState<Nullable<number>>(null)
+
+  useEffect(() => {
+    if (encodedUserId && encodedProfileUserId) {
+      if (tag === SUPPORTING_USER_LIST_TAG) {
+        const profileSupporting = supportingMap[encodedProfileUserId] ?? {}
+        const supporting = profileSupporting[encodedUserId] ?? {}
+        setAmount(supporting.amount ?? null)
+      } else if (tag === TOP_SUPPORTERS_USER_LIST_TAG) {
+        const profileSupporters = supportersMap[encodedProfileUserId] ?? {}
+        const supporter = profileSupporters[encodedUserId] ?? {}
+        setRank(supporter.rank ?? null)
+        setAmount(supporter.amount ?? null)
+      }
+    }
+  }, [
+    encodedUserId,
+    encodedProfileUserId,
+    supportingMap,
+    supportersMap,
+    tag,
+    userId
+  ])
+
+  return TIP_SUPPORT_TAGS.has(tag) ? (
+    <div className={styles.tipContainer}>
+      {TOP_SUPPORTERS_USER_LIST_TAG === tag ? (
```

Comment:
Rather than all the if/else-if logic in this component, would it make more sense to have two separate components composed together here?